pipeline/git_test.go

```diff
@@ -9,16 +9,13 @@ import (
 
 func TestGitCloneRepo(t *testing.T) {
 	repo := &gaia.GitRepo{
-		URL: "https://github.com/gaia-pipeline/gaia",
+		URL:       "https://github.com/gaia-pipeline/gaia",
```

Comment:
Consider creating a test repo under `gaia-pipeline` so that it checks out that instead of the entire gaia repo which will potentially be large at some point. Also, this test right now takes about 8 seconds which isn't very nice. Create a test repo with a README in it in order to verify this functionality.

Also, `LocalDest` wasn't defined and `RemoveAll` wasn't called with defer, which would end up as being not cleaned if `gitCloneRepo` would have failed.